bin/pasta_file_change_recipients.py

```diff
@@ -0,0 +1,108 @@
+"""
+PaStA - Patch Stack Analysis
+
+Author:
+  Shubhamkumar Pandey <b18194@students.iitmandi.ac.in>
+
+This work is licensed under the terms of the GNU GPL, version 2.  See
+the COPYING file in the top-level directory.
+"""
+
+import os
+import sys
+import argparse
+
+from tqdm import tqdm
+from collections import defaultdict, Counter
+from logging import getLogger
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+from pypasta import *
+from pypasta.LinuxMailCharacteristics import email_get_recipients
+from pypasta.Repository.Patch import Diff
+
+log = getLogger(__name__[-15:])
+
+RECIPIENTS = None
+
+def dump_csv(config, f_recipients):
+    repo = config.repo
+    global RECIPIENTS
+    RECIPIENTS = defaultdict(Counter)
+    for message_id in tqdm(repo.mbox.get_ids(time_window=config.mbox_time_window)):
+        patch = repo[message_id]
+        mail = repo.mbox.get_messages(message_id)[0]
+        recipients = email_get_recipients(mail)
+        for filename in patch.diff.affected:
+            for recipient in recipients:
+                RECIPIENTS[filename][recipient] += 1
+    
+    if f_recipients:
+        with open(f_recipients, 'w') as f:
+            f.write('filename, recipients, count\n')
+            for filename, ctr in RECIPIENTS.items():
+                ctr = sorted(ctr.items(), key = lambda x: x[1], reverse=True)
+                f.write(f"{filename}, {','.join(map(lambda x: x[0]+','+str(x[1]),ctr))}\n")
```

Comment:
Uh, black magic! Why don't you simply use python-pickle to dump RECIPIENTS?

Reply:
@rralf  I asked you before for that for introducing a cache parameter in the config object. I saw the how pkl files are handled in the system and thought to introduce f_recipient_pkl in config object, but it seemed to be wrong to touch Config. So do I make a pkl file inside resource directory along with like patch-group and other caches without making an attribute inside Config?

Reply:
Simply choose a constant filename for the moment. We can later introduce a proper name when it comes to integration.

Reply:
I did, I also thought that would be better for now. So I did it that way in https://github.com/lfd/PaStA/pull/74, placed it with other pkl files in resource directory.